grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/GraqlTraversal.java

```diff
@@ -89,6 +91,17 @@ static GraqlTraversal create(Set<? extends List<Fragment>> fragments) {
     public abstract ImmutableSet<ImmutableList<Fragment>> fragments();
 
     /**
+     * @param conceptMap map defining mappings between concept ids
+     * @return graql traversal with concept id transformed according to the provided concept map
+     */
+    public GraqlTraversal transform(Map<ConceptId, ConceptId> conceptMap){
+        ImmutableList<Fragment> fragments = ImmutableList.copyOf(
+                Iterables.getOnlyElement(fragments()).stream().map(f -> f.transform(conceptMap)).collect(Collectors.toList())
+        );
+        return new AutoValue_GraqlTraversal(ImmutableSet.of(fragments));
+    }
```

Comment:
This won't work if the query has a disjunction. Is that intended?

Reply:
all disjunctions are DNFed to sets of conjunctions on entry in `MatchInfer`